i2c/sample-implementations/Atmel_SAMDR_series/sensirion_i2c_ASF.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2018, Sensirion AG
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Sensirion AG nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include "delay.h"
+#include "asf.h"
+
+
+/*
+*   Fixed I2C on SAMR architectures, this code its done to work with ASF from Microchip.
+*   15/11/2021
+*   On the i2c init, you should provide the CONF_MASTER_SDA_PINMUX and CONF_MASTER_SCK_PINMUX pin configuration.
+*   Then, we should configure the SERCOM module to use, with the CONF_I2C_MASTER_MODULE parameter, an example configuration for the ATSAMDR34J18 could be like:
+*   
+*   #define CONF_I2C_MASTER_MODULE    SERCOM3
+*   #define CONF_MASTER_SDA_PINMUX    PINMUX_PA16D_SERCOM3_PAD0
+*   #define CONF_MASTER_SCK_PINMUX    PINMUX_PA17D_SERCOM3_PAD1
+*   #define TIMEOUT 1000
```

Comment:
Can you add
```
#ifndef CONF_I2C_MASTER_MODULE
#define CONF_I2C_MASTER_MODULE SERCOM3
#endif
```
for all the defines? That way it compiles by default, but a user can still override the defines.